react/src/tests/opReturn.test.ts

```diff
@@ -1,82 +1,122 @@
-    expect(prependPaymentIdWithPushdata(paymentId)).toBe('2d' + paymentId)
-  })
-  it ('Doesnt throw at limit', () => {
-    expect(prependPaymentIdWithPushdata('aa'.repeat(75))).toBe('4b' + 'aa'.repeat(75))
-  })
-  it ('Throws if too big', () => {
-    expect(() =>
-      prependPaymentIdWithPushdata('aa'.repeat(76))
-    ).toThrow('Maximum 75 byte size exceeded for paymentId: 76')
-  })
-})
+  it('Prepends 3 bytes paymentId with pushdata', () => {
+    expect(prependPaymentIdWithPushdata('aa00bb')).toBe('03aa00bb');
+  });
+  it('Prepends 45 bytes paymentId with pushdata', () => {
+    const paymentId =
+      'ff019349025bc95aefa6dfab756f6ea7cb96eadbfa696abd816abef6a97fbad8917afd5bbdf6a7b5976b95a6dc';
+    expect(prependPaymentIdWithPushdata(paymentId)).toBe('2d' + paymentId);
+  });
+  it('Prepends paymentId with pushdata at 75 bytes limit', () => {
+    expect(prependPaymentIdWithPushdata('aa'.repeat(75))).toBe(
+      '4b' + 'aa'.repeat(75),
+    );
+  });
+  it('Throws if paymentId greater than 75 bytes limit', () => {
+    expect(() => prependPaymentIdWithPushdata('aa'.repeat(76))).toThrow(
+      'Maximum 75 byte size exceeded for paymentId: 76',
+    );
+  });
+  it('Throws if paymentId not even in length', () => {
+    expect(() => prependPaymentIdWithPushdata('abc')).toThrow(
+      'Malformed input; paymentId hex should never be of odd length',
+    );
+  });
+});
 
 describe('generatePushdataPrefixedPaymentId', () => {
-  it ('8 byte paymentId', () => {
-    const prefixedPaymentId = generatePushdataPrefixedPaymentId(8)
-    expect(prefixedPaymentId.length).toBe(18) // 8 * 2 (two hex chars=byte) + 2(pushdata prefix)
-    expect(prefixedPaymentId.slice(0,2)).toBe('08')
-  })
-  it ('16 byte paymentId', () => {
-    const prefixedPaymentId = generatePushdataPrefixedPaymentId(16)
-    expect(prefixedPaymentId.length).toBe(34) // 16 * 2 + 2
-    expect(prefixedPaymentId.slice(0,2)).toBe('10')
-  })
-  it ('75 byte paymentId', () => {
-    const prefixedPaymentId = generatePushdataPrefixedPaymentId(75)
-    expect(prefixedPaymentId.length).toBe(152) // 75 * 2 + 2
-    expect(prefixedPaymentId.slice(0,2)).toBe('4b')
-  })
-  it ('fails at 76 byte paymentId', () => {
-    expect(() =>
-      generatePushdataPrefixedPaymentId(76)
-    ).toThrow('Maximum 75 byte size exceeded for paymentId: 76')
-  })
-})
+  it('Generates 8 byte paymentId', () => {
+    const prefixedPaymentId = generatePushdataPrefixedPaymentId(8);
+    expect(prefixedPaymentId.length).toBe(18); // 8 * 2 (two hex chars=byte) + 2(pushdata prefix)
+    expect(prefixedPaymentId.slice(0, 2)).toBe('08');
+  });
+  it('Generates 16 byte paymentId', () => {
+    const prefixedPaymentId = generatePushdataPrefixedPaymentId(16);
+    expect(prefixedPaymentId.length).toBe(34); // 16 * 2 + 2
+    expect(prefixedPaymentId.slice(0, 2)).toBe('10');
+  });
+  it('Generates paymentId at 75 bytes limit', () => {
```

Comment:
isn't the paymentId always 8 bytes, to spec? 

Does the function support this as variable because this could change in the future?

Could just hardcode it in the function and have only 1 or 2 unit tests